urts/src/lib.rs

```diff
@@ -2,22 +2,57 @@
 
 #![doc = include_str!("../README.md")]
 
-use mc_sgx_core_sys_types::sgx_status_t;
-use mc_sgx_urts_sys::{sgx_create_enclave_from_buffer_ex, sgx_destroy_enclave};
-use mc_sgx_urts_sys_types::sgx_enclave_id_t;
-use std::{ops::Deref, os::raw::c_int, ptr};
+use mc_sgx_core_types::{Error, TargetInfo};
+use mc_sgx_urts_sys::{
+    sgx_create_enclave_from_buffer_ex, sgx_destroy_enclave, sgx_get_target_info,
+    SGX_CREATE_ENCLAVE_EX_KSS, SGX_CREATE_ENCLAVE_EX_KSS_BIT_IDX, SGX_CREATE_ENCLAVE_EX_PCL,
+    SGX_CREATE_ENCLAVE_EX_PCL_BIT_IDX,
+};
+use mc_sgx_urts_sys_types::{sgx_enclave_id_t, sgx_kss_config_t};
+use mc_sgx_util::ResultInto;
+use std::{ffi::c_void, fs::File, io::Read, mem::MaybeUninit, os::raw::c_int, path::Path, ptr};
 
-#[derive(Debug, PartialEq, Eq)]
-pub enum Error {
-    // An error provided from the SGX SDK
-    SgxStatus(sgx_status_t),
+/// Structure defining configuration for Key Sharing and Separation
+pub struct KssConfig {
+    pub config_id: [u8; 64],
+    pub config_svn: u16,
+}
+
+// We can't derive Default because Default isn't implemented for [u8; 64] in
+// current Rust
+impl Default for KssConfig {
+    fn default() -> Self {
+        // There are no restrictions on these values, so use 0 as default
+        KssConfig {
+            config_id: [0; 64],
+            config_svn: 0,
+        }
+    }
+}
+
+impl From<KssConfig> for sgx_kss_config_t {
+    fn from(input: KssConfig) -> sgx_kss_config_t {
+        sgx_kss_config_t {
+            config_id: input.config_id,
+            config_svn: input.config_svn,
+        }
+    }
+}
+
+impl From<sgx_kss_config_t> for KssConfig {
+    fn from(input: sgx_kss_config_t) -> KssConfig {
+        KssConfig {
+            config_id: input.config_id,
+            config_svn: input.config_svn,
+        }
+    }
 }
```

Comment:
I think we should do the newtype wrapper and `KssConfig::new(config_id: ConfigId, config_svn: ConfigSvn)`, under the assumption that there isn't a particular use for mutable accessors...

Reply:
OK, I'll do that.

For posterity - the reason I did this originally was because I figured there _was_ a use for mutable accessors, and the idiomatic Rust way to do that is just to make the fields public rather than mess around with `set_config(&mut self, config: ConfigId)` functions. James' comment about keeping them in sync is a better argument for the newtype though.